Release/1.9.0 (#209)
fivetran/dbt_hubspot: CHANGELOG.md

Reviewer: I'm not sure why, but it appears your changes overwrote the entire Changelog with a new entry, is it possible to leverage Claude to revert these changes so that we only see the new version updates, to confirm that no changes were made to prior changelog entries?

Author: Done!

| `int_hubspot__daily_deal_history`<br>`int_hubspot__deal_calendar_spine`<br>`int_hubspot__pivot_daily_deal_history`<br>`int_hubspot__scd_daily_deal_history`<br>`int_hubspot__daily_contact_history`<br>`int_hubspot__contact_calendar_spine`<br>`int_hubspot__pivot_daily_contact_history`<br>`int_hubspot__scd_daily_contact_history`<br>`int_hubspot__daily_company_history`<br>`int_hubspot__company_calendar_spine`<br>`int_hubspot__pivot_daily_company_history`<br>`int_hubspot__scd_daily_company_history` | New intermediate models | | | Used to power new daily property history end models. |
| `stg_hubspot__deal_property_history`<br>`stg_hubspot__company_property_history`<br>`stg_hubspot__contact_property_history` | New staging columns | | `_fivetran_start`, `_fivetran_end`, `_fivetran_active` | SCD metadata fields used to power daily history models. |

## Feature Updates

### Property History Models

The following changes apply to daily property history models:

> - Adds the `hubspot__contact_property_history_columns` variable to pivot out additional properties as `string` columns in `hubspot__daily_contact_history`. See the [README](https://github.com/fivetran/dbt_hubspot/tree/main?tab=readme-ov-file#daily-contact-ticket-and-deal-history) for configuration details.

Reviewer: ```suggestion
- Adds the `hubspot__contact_property_history_columns` variable to pivot out additional properties as `string` columns in `hubspot__daily_contact_history`. See the [README](#daily-contact-ticket-deal-and-company-history ) for configuration details.
```

Author: Hm I think we need the link but I will update

Reviewer: Oh whoops. I think you're right.

> - Adds the `hubspot__deal_property_history_columns` variable to pivot out additional properties as `string` columns in `hubspot__daily_deal_history`. See the [README](https://github.com/fivetran/dbt_hubspot/tree/main?tab=readme-ov-file#daily-contact-ticket-and-deal-history) for configuration details.

Reviewer: ```suggestion
- Adds the `hubspot__deal_property_history_columns` variable to pivot out additional properties as `string` columns in `hubspot__daily_deal_history`. See the [README](https://github.com/fivetran/dbt_hubspot/tree/main?tab=readme-ov-file##daily-contact-ticket-deal-and-company-history) for configuration details.
```

Author: Updated

- Adds the `hubspot__company_property_history_columns` variable to pivot out additional properties as `string` columns in `hubspot__daily_company_history`. See the [README](https://github.com/fivetran/dbt_hubspot/tree/main?tab=readme-ov-file#daily-contact-ticket-deal-and-company-history) for configuration details.

> - Adds the `deal_history_extension_days` variable to limit or extend the history of deals in `hubspot__daily_deal_history` past their close dates (default=30 days). See the [README](https://github.com/fivetran/dbt_hubspot/tree/main?tab=readme-ov-file#extending-ticket-and-deal-history-past-closing-date) for configuration details.

Reviewer: ```suggestion
- Adds the `deal_history_extension_days` variable to limit or extend the history of deals in `hubspot__daily_deal_history` past their close dates (default=30 days). See the [README](https://github.com/fivetran/dbt_hubspot/tree/main?tab=readme-ov-file#extending-ticket-and-deal-history) for configuration details.
```

Author: Updated

- Adds the `hubspot__contact_list_pass_through_columns` variable to bring additional columns from the `contact_list` source table through to `hubspot__contact_lists` and `stg_hubspot__contact_list`. See the [README](https://github.com/fivetran/dbt_hubspot/tree/main?tab=readme-ov-file#include-passthrough-columns) for configuration details.
- The [`hubspot__pass_through_all_columns`](https://github.com/fivetran/dbt_hubspot/#include-passthrough-columns) and [`hubspot__enable_all_property_labels`](https://github.com/fivetran/dbt_hubspot/#adding-property-label) variables now also apply to `company_list` and `contact_list` models.

## Under the Hood
- Casts `form_id` and/or `conversion_id` as strings when missing from source data in `stg_hubspot__form` and `stg_hubspot__contact_form_submission`.
- Ensures `user_id` is an integer in `stg_hubspot__users` when missing from source data.

> - Optimizes `int_hubspot__ticket_calendar_spine` by limiting how far into the future it looks (one day vs one week) and its start date on incremental runs.

Reviewer: I'm actually realizing there's a potential issue for customers on `hubspot__daily_ticket_history` using `ticket_history_extension_days` previously. Materialized rows for days 2-7 in the future will fall outside the spine window on incremental rows and will require a full refresh. Since it's not a standard case, should it be added to the  schema/data section, or just noted here?

Author: I'll note here

Reviewer: I think this was added below but it's not quite clear it applies to this bullet, so recommend consolidating.

- Adds the `hubspot__contact_list_pass_through_columns` variable to bring additional columns from the `contact_list` source table through to `hubspot__contact_lists` and `stg_hubspot__contact_list`. See the [README](https://github.com/fivetran/dbt_hubspot/tree/main?tab=readme-ov-file#include-passthrough-columns) for configuration details.
- The [`hubspot__pass_through_all_columns`](https://github.com/fivetran/dbt_hubspot/#include-passthrough-columns) and [`hubspot__enable_all_property_labels`](https://github.com/fivetran/dbt_hubspot/#adding-property-label) variables now also apply to `company_list` and `contact_list` models.

## Under the Hood
- Casts `form_id` and/or `conversion_id` as strings when missing from source data in `stg_hubspot__form` and `stg_hubspot__contact_form_submission`.
- Ensures `user_id` is an integer in `stg_hubspot__users` when missing from source data.
- Optimizes `int_hubspot__ticket_calendar_spine` by limiting how far into the future it looks (one day vs one week) and its start date on incremental runs.

> > **This will require a `--full-refresh` run**.

Reviewer: It's unclear what this is in reference to since it's spaced out, should we add this at the end of line 46 or a subbullet in line 47?

Author: I'll make it a subbullet

| [`hubspot__marketing_event_performance`](https://fivetran.github.io/dbt_hubspot/#!/model/model.hubspot.hubspot__marketing_event_performance) | New end model | | | Each record represents a marketing event with aggregated attendance metrics, contact-level participation counts, segment list membership, and optional pivoted custom properties. |
| [`hubspot__contacts`](https://fivetran.github.io/dbt_hubspot/#!/model/model.hubspot.hubspot__contacts) | New fields | | `calculated_first_conversion_fields_responded_to`<br>`calculated_most_recent_conversion_fields_responded_to`<br>`calculated_first_conversion_total_responses`<br>`calculated_most_recent_conversion_total_responses` | Comma-separated list and count of fields submitted during the contact's first and most recent form conversions. |

> | `stg_hubspot__conversation_thread`<br>`stg_hubspot__conversation_thread_tmp`<br>`stg_hubspot__conversation_message_history`<br>`stg_hubspot__conversation_message_history_tmp`<br>`stg_hubspot__conversation_message_recipient`<br>`stg_hubspot__conversation_message_recipient_tmp`<br>`stg_hubspot__conversation_actor`<br>`stg_hubspot__conversation_actor_tmp`<br>`stg_hubspot__conversation_channel`<br>`stg_hubspot__conversation_channel_tmp`<br>`stg_hubspot__conversation_channel_account`<br>`stg_hubspot__conversation_channel_account_tmp`<br>`stg_hubspot__conversation_inbox`<br>`stg_hubspot__conversation_inbox_tmp`<br>`stg_hubspot__submission_response`<br>`stg_hubspot__submission_response_tmp`<br>`stg_hubspot__company_list`<br>`stg_hubspot__company_list_tmp`<br>`stg_hubspot__company_list_member`<br>`stg_hubspot__company_list_member_tmp`<br>`stg_hubspot__marketing_event`<br>`stg_hubspot__marketing_event_tmp`<br>`stg_hubspot__marketing_event_list`<br>`stg_hubspot__marketing_event_list_tmp`<br>`stg_hubspot__marketing_event_participant`<br>`stg_hubspot__marketing_event_participant_tmp`<br>`stg_hubspot__marketing_event_custom_property`<br>`stg_hubspot__marketing_event_custom_property_tmp` | New staging models | | | Used to power the above new end models. |

Reviewer: I'm wondering if we should list the staging models in one row and the tmp models in another to prevent this entry from being too heavy to read.

Reviewer: Or maybe something like 
```
`stg_hubspot__conversation_thread` (and equivalent `*_tmp` model)<br>`
` New staging and tmp models`
```

Author: Adjusted, lemme know what you think

> | `int_hubspot__daily_deal_history`<br>`int_hubspot__deal_calendar_spine`<br>`int_hubspot__pivot_daily_deal_history`<br>`int_hubspot__scd_daily_deal_history`<br>`int_hubspot__daily_contact_history`<br>`int_hubspot__contact_calendar_spine`<br>`int_hubspot__pivot_daily_contact_history`<br>`int_hubspot__scd_daily_contact_history`<br>`int_hubspot__daily_company_history`<br>`int_hubspot__company_calendar_spine`<br>`int_hubspot__pivot_daily_company_history`<br>`int_hubspot__scd_daily_company_history` | New intermediate models | | | Used to power new daily property history end models. |

Reviewer: The intermediate models are all materialized as ephemeral in the `sales`/`marketing` folders, so they wouldn't be visible to customers unless they change the materialization. Should we remove this line from the changelog.

Author: These ones are all incremental and have their own config blocks

> | `stg_hubspot__deal_property_history`<br>`stg_hubspot__company_property_history`<br>`stg_hubspot__contact_property_history` | New staging columns | | `_fivetran_start`, `_fivetran_end`, `_fivetran_active` | SCD metadata fields used to power `hubspot__daily_deal_history`. |

Reviewer: This also powers the other daily history models too:

```suggestion
| `stg_hubspot__deal_property_history`<br>`stg_hubspot__company_property_history`<br>`stg_hubspot__contact_property_history` | New staging columns | | `_fivetran_start`, `_fivetran_end`, `_fivetran_active` | SCD metadata fields used to power `hubspot__daily_company_history`, `hubspot__daily_contact_history`, `hubspot__daily_deal_history`. |
```

Author: Adjusted

- The [`hubspot__pass_through_all_columns`](https://github.com/fivetran/dbt_hubspot/#include-passthrough-columns) and [`hubspot__enable_all_property_labels`](https://github.com/fivetran/dbt_hubspot/#adding-property-label) variables now also apply to `company_list` and `contact_list` models.

### Enabling/Disabling New Models

The following changes allow for disabling newly added transformations in self-managed dbt projects:

- Adds the `hubspot_conversation_enabled` variable to support enabling/disabling `conversation` models. Dynamically configured in Quickstart; `true` by default otherwise. 
- Adds the `hubspot_submission_response_enabled` variable to support enabling/disabling `submission_response` models. Dynamically configured in Quickstart; `true` by default otherwise.
- Adds the `hubspot_company_list_enabled` and `hubspot_company_list_member_enabled` variables to support enabling/disabling the new `company_list` and `company_list_member` models. Both are dynamically configured in Quickstart; enabled by default when `hubspot_sales_enabled` is `true` otherwise.

> - Adds the `hubspot_marketing_event_enabled`, `hubspot_marketing_event_list_enabled`, `hubspot_marketing_event_participant_enabled`, and `hubspot_marketing_event_custom_property_enabled` variables to support enabling/disabling the new marketing event models and their source tables. All default to `true` when `hubspot_marketing_enabled` is `true`.

Reviewer: Should we consolidate and add condensed versions of these bullets to each corresponding notes section to each end model? Also we should link to the relevant README section.

Author: Good idea -- adding

## Under the Hood

> - Casts `form_id` and `conversion_id` as strings when missing from source data in `stg_hubspot__form` and `stg_hubspot__contact_form_submission`.

Reviewer: Slight rewrite

```suggestion
- Casts `form_id` as strings when missing from source data in `stg_hubspot__form`, as well as for `form_id` and `conversion_id` in `stg_hubspot__contact_form_submission`.
```

Author: That's a bit clunky, what if I adjust to `form_id and/or conversion_id` so it's not totally implied that both fields exist in both models? I don't think we need to get very detailed, but otherwise I could split into two different bullets
